test.py

```diff
@@ -189,18 +189,22 @@ def test_new_it_guy(edap):
         edap.make_uid_member_of("blbec", it_group_dn)
 
 
-def test_czech_franchise_becomes_present(edap):
-    assert not edap.subobject_exists_at(f"cn=cz_prg,{edap.FRANCHISES}", "posixGroup")
-    edap.create_franchise("cz_prg")
-    assert edap.subobject_exists_at(f"cn=cz_prg,{edap.FRANCHISES}", "posixGroup")
-    assert edap.object_exists(f"&(commonName=cz_prg)(description=Czech Republic)", "posixGroup")
-
-
-def test_label_franchise(edap):
-    assert edap.label_franchise("cz") == "Czech Republic"
-    assert edap.label_franchise("cz_prg") == "Czech Republic"
+def test_czech_country_becomes_present(edap):
+    country_cname = 'cz_prg'
+    country_description = 'Czech Republic'
+    assert not edap.subobject_exists_at(f"cn={country_cname},{edap.COUNTRIES}", "posixGroup")
+    edap.create_country(country_cname)
+    assert edap.subobject_exists_at(f"cn={country_cname},{edap.COUNTRIES}", "posixGroup")
+    assert edap.object_exists(f"&(commonName={country_cname})(description={country_description})", "posixGroup")
+    assert edap.get_country(country_cname)['cn'][0] == country_cname.encode('utf-8')
+    assert edap.get_country(country_cname)['description'][0] == country_description.encode('utf-8')
+
+
+def test_label_country(edap):
+    assert edap.label_country("cz") == "Czech Republic"
+    assert edap.label_country("cz_prg") == "Czech Republic"
```

Comment:
`cz_prg` is not a country code. I think that it should pass, so the label function should allow a franchise code in form of `<country_code>` or `<country_code>_<something>`, i.e. `cz` and `cz_prg` are OK, but `cz_` is not OK. I guess that you can call `'cz_prg'.split('_', 1)`, and the first split element has to be the ISO code, while the second split element has to be non-empty if it exists.

Reply:
I did `splitted_code = code.split('_')` and checking if `splitted_code[1]` exists, to reject `cz___` as well.